### PR TITLE
Initial pipeline implementation

### PR DESCRIPTION
**Pull request by vincerubinetti**
This is meant to be a basic vertical slice of the tech stack. Goals: flexible scraping, modular organization, type-safe from end to end, flexible data querying.

Right now the size of the data is very small and having an actual db might is probably overkill. I suppose it's good in that, if we're requested to get new data, anyone familiar with SQL could do it very quickly. It also may benefit us if the data grows to be an order of magnitude larger at some point, for more performant querying and even perhaps having a continuously running server that can get things dynamically (e.g. a more interactive dashboard). But if neither of these things happen, we may want to downgrade to just a simple `.json` file later.

See the related issues that this closes. This does not ingest e.g. Scimago citation information, but it does ingest iCite as a proof of concept. The actual contents of the webpage/report it generates is likely not useful (and doesn't explore the full depth of data ingested and integrated into the database), but again is just there as a proof of concept. In the next PR, I will focus on making an actual useful report. Casey suggested starting with per-core-project reports and a cfde-global report that list associated publications in order of date (and I guess some other high level info).

For now, the app is being deployed via Netlify (production version will be https://cfde-eval.netlify.app/ after this PR is merged).

High level description of changes, roughly in order of diff:

- github actions "compile" workflow to manually run this pipeline
- readmes in each sub folders as appropriate
- /app
  - vue app for creating the dashboard and pdf. vite, vue 3, composition api, typescript, eslint, prettier
  - pie chart component with apache echarts
- /data 
  - collection of node scripts run via tsx
  - type definitions for third party apis generated automatically
  - top-level "compile" script that runs high-level steps
  - sqlite database run by `better-sqlite3` library and interacted with via Kysely for type safety and query-constructing ergonomics
  - individual db tables for funding opportunities, projects/grants, and publications. to start, i just kept fields from third party apis that i thought were useful. schemas defined via ts.
  - report function that queries db to collate high-level data for webapp to display. exports it to json, runs webapp, and prints to pdf report. pdf report then deployed with webapp for public access.
  - funding opportunities scraped from cfde website html/pdf documents using playwright
  - projects/grant records obtained from reporter api
  - associated publications obtained from reported api, and supplemented with extra citation data from icite
  - more ingest sources to come later
  - bunch of useful util functions
    - pretty logging (will be important for debugging when the pipeline has more and more complex steps)
    - custom memoize func to avoid hitting apis too much (and possibly be rate limited)

### CODE REVIEW

File: app/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_TITLE="CFDE Evaluation"
```
Comment (cgreene) [inline app/.env:1]:
```suggestion
VITE_TITLE="CFDE Evaluation Core Dashboard"
```

File: app/.env [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+VITE_DESCRIPTION="Metrics for evaluating NIH Common Fund Data Ecosystem projects"
```
Comment (cgreene) [inline app/.env:2]:
```suggestion
VITE_DESCRIPTION="Evaluation Core dashboard to support the NIH Common Fund Data Ecosystem"
```

**Review by cgreene — COMMENTED** [review]
If I understand the strategy correctly, we're storing the database in the repository. Can we update the README for the repository to broadly describe the structure/technial approach and note some caveats (e.g., I assume we should be clear not to store tokens, etc. in the database). Instead, those should presumably be stored in files protected with something like git crypt.

**Comment by cgreene** [comment]
Somewhere on the page (maybe footer?) can we write "Developed with support from NIH Award U54 OD036472"

**Comment by vincerubinetti** [comment]
@cgreene I believe I've addressed your comments in the last few commits. Sean is out. I'd like to merge and continue working on other discussed features.

**Pull request merged**